### ISSUE TITLE
client_sw unable to copy from local system and install on remote as root

### ISSUE THREAD

**Issue by pboushy** [OneIdentity/ansible-authentication-services#40]
I have a pretty basic playbook:
```
---
- name: App (Redhat)
  hosts: app
  vars:
    client_sw_dir: "./files/vas"
    client_sw_tmp_dir: /tmp/vas
    client_sw_pkg_state:
      vasclnts: present
      dnsupdate: present
    client_sw_facts_generate: true
    client_sw_facts_verbose: true
  roles:
    - name: base_linux
    - name: oneidentity.authentication_services.client_sw
    - name: app
```

When this runs, it routinely fails on `client_sw` on `install vasclnts` with an error:
```
warning: /tmp/vas/ansible-as-client_sw/vasclnts-5.1.2-8000.x86_64.rpm: Header V4 RSA/SHA256 Signature, key ID 4417450c: NOKEY\nerror: can't create transaction lock on /var/lib/rpm/.rpm.lock (Permission denied)
```

If I add `ansible_become: true` under vars, then it hangs on `software install package directory check`
Running with `-vvv` tells me that it's trying to run this command:
```
EXEC /bin/sh -c 'sudo -H -S -p "[sudo via ansible, key=REDACTED] password:" -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-REDACTED ; /usr/local/Cellar/ansible/8.2.0_2/libexec/bin/python /Users/pboushy/.ansible/tmp/ansible-tmp-1699086122.439269-52181-1725620806624/AnsiballZ_client_sw_pkgs.py'"'"' && sleep 0'
```
But the become password on my local computer is vastly different than the one on the remote.
I would expect that the software install package directory check would NOT attempt sudo.

It appears that either `become: false` needs to be added to tasks that happen locally, or `become: true` needs to be added in places that require root (i.e. installing sw)

Are there any workarounds? Or did I miss something in documentation that would resolve this?

**Comment by pboushy** [comment]
FYI - report creation on 127.0.0.1 also fails if you include `become: true`

**Comment by llnagy76** [comment]
Hi Phillip,
Thank you very much for your bug report (and enhancement request regarding keytab)! At the moment it looks like I can start working on these next week.

**Comment by pboushy** [comment]
That's great to hear. Let me know when you think you've got it and I'll test a branch before you merge it in if you want.

FYI - I was running `ansible-playbook` version 2.15.5 from macOS 13.5 connecting to a redhat 8.7 server using the below command:
```
ansible-playbook -i "environments/${environment}" -u "$username" -K --ask-vault-pass playbook.yaml
```
The Mac has no credentials that match the redhat server. Connection to the redhat server was using an SSH key and a become password passed (`-K` argument) 

Not sure that any of that matters, but if you wanted to test locally it gives you some info on the environment I experienced this in.

**Comment by llnagy76** [comment]
@pboushy 
Thank you for your offer of testing my fix before merging it in!
I would like to let you know that I have managed to reproduce your issue in my test environment.

**Comment by llnagy76** [comment]
@pboushy
I have pushed my changes to a new branch:
https://github.com/llnagy76/ansible-authentication-services/tree/always_run_commands_on_localhost_without_sudo
Please test it in your environment! You need to add `ansible_become: true` to your playbook.
Thank you very much for your help!

I have changed only role client_sw. The bug you found affects all roles, but I would wait your test results before modifying all roles.

**Comment by llnagy76** [comment]
The fix has been released in version 0.3.0.
